### DISCUSSION TITLE
Override Stop command

### DISCUSSION THREAD

**Discussion by tiago19fp** [eclipse/mosaic#321] [Q&A] [answered]
Hello,

Is it possible to cancel a command to stop a vehicle in a certain position already called? For example, if a flag was activated the vehicle would no longer  stop at the position.
To stop the vehicle I use the following code:

IRoadPosition roadPosition = RoadPositionFactory.createAtEndOfRoute( Collections.singletonList(getOs().getRoadPosition().getConnection().getId()), 0);
getOs().stop(roadPosition, STOP, time_to_stop * TIME.SECOND);

**Comment by kschrab** [comment, marked as answer]
To cancel a previously issued stop, you'd need to call the stop command again with the same (equal) road position but with a time of 0.

**Reply by schwepmo** [reply]
I feel like, `getOs().resume()` should also work 🤞

**Reply by tiago19fp** [reply]
> I feel like, `getOs().resume()` should also work crossed_fingers

I tried to use  `getOs().resume()` but it didn't work, the vehicle kept stopping at the defined roadPositon. I used  what @kschrab  recommended and it worked.

**Reply by schwepmo** [reply]
Okay, thanks for trying this for us! Maybe we need to add this to the backlog 😅 
As a follow-up question did you call `resume()` after the vehicle already stopped at the stopping position or beforehand?

**Reply by tiago19fp** [reply]
No problem, happy to help! :+1: 
I called the `resume()` command before the vehicle stopped at the target position.

**Comment by fabi-kienzle** [comment]
Hi @kschrab and @schwepmo, I have a similar Issue as I want to override the changeSpeedWithInterval command. 

Scenario:
- With a first command, I set the speed to 0 with a certain duration. 
- Within that duration, I want to override to a different higher speed with a new duration. 
- After the second speed adjustment, the vehicle starts to accelerate (as I would expect) until the duration of the first command runs out and then suddenly drops to 0. 

Is there a possibility to fully override or cancel the first changeSpeedWithInterval command, so that it does not interfere with new comands?
I also tried calling the command with a time of 0 and also resetting the Speed (resetSpeed()) but that didn't work.

Thanks in advance! :)

**Reply by kschrab** [reply]
@fabi-kienzle Apparently it is currently not possible to override previous `changeSpeedWithInterval` calls, if the duration of that previous call has not yet passed. Have you tried to use `slowDown(speed, duration)` instead of `changeSpeedWithInterval`? (at least for the first call)
